### PR TITLE
Updates/modal updates

### PR DESCRIPTION
**Pull request by zepfan**
- Made the mobile modal not full screen by default, instead gave it a `full` prop
- Added some styles for "text buttons" 
- Removed an old addition that wasn't needed

### CODE REVIEW

File: src/elements/Button/Button.js [hunk 1]
```diff
@@ -61,6 +62,8 @@ Button.propTypes = {
 	outline: PropTypes.bool,
 	/** Bare style */
 	bare: PropTypes.bool,
+	/** Text button styles */
+	text: PropTypes.bool,
```
Comment (jfurfaro) [inline src/elements/Button/Button.js]: if we have a text-only button... is it still a button? What's the use case vs. an anchor with different styling?

File: src/elements/Modal/Modal.js [hunk 2]
```diff
@@ -17,7 +17,8 @@ class Modal extends Component {
 		dequeueModal: PropTypes.func.isRequired,
 		modalRoot: PropTypes.instanceOf(window.Element).isRequired,
 		className: PropTypes.string,
-		dismissible: PropTypes.bool
+		dismissible: PropTypes.bool,
+		full: PropTypes.bool
```
Comment (jfurfaro) [inline src/elements/Modal/Modal.js]: if we're going to have this be a toggle, we should name it appropriately `fullScreenMobile` or similar to disambiguate from a full screen modal at any screen size

File: src/elements/Button/Button.js (hunk 1 above)
Comment (zepfan) [reply src/elements/Button/Button.js]: So it's for buttons like these:
screen shot 2018-10-30 at 2 32 28 pm

Which trigger actions, modals, etc. I was considering just using anchors at first, but these really do fit the bill as buttons more. But the variant style needed to be more of just a straight, "link style" button w/o all of the padding, background colors, etc. Maybe `text` isn't the best descriptor for it though

Comment (zepfan) [reply src/elements/Button/Button.js]: Or, maybe this would make sore sense as an `IconButton`?

Comment (jfurfaro) [reply src/elements/Button/Button.js]: I thought the `bare` button covered this? The icons I would definitely say just go inside with the text, so `IconButton` wouldn't make sense here.

```
<Button bare><Icon ...> Click me</Button>
```

File: src/elements/Modal/Modal.js (hunk 2 above)
Comment (zepfan) [reply src/elements/Modal/Modal.js]: So it's not actually just for mobile, you can make it fullscreen for all of it (or, if you wanted it to be specifically for one or the other, you could use it conjunction with `isMobile()`). But maybe `fullScreen` is still a better name for it?

Comment (jfurfaro) [reply src/elements/Modal/Modal.js]: A full screen modal on desktop isn't a modal, it's an overlay. I would have this switch only for mobile, if needed.

I would agree with an argument to never have modals be full screen, and if we need an overlay just for mobile we would create and use an `Overlay` ui component

Comment (zepfan) [reply src/elements/Modal/Modal.js]: Alright, I've made the `fullScreen` prop only work for mobile

File: src/elements/Button/Button.js (hunk 1 above)
Comment (zepfan) [reply src/elements/Button/Button.js]: The `bare` style was kind of a different thing, but we don't actually use it anywhere / it's never appeared in any designs as it was. So yeah -- I've pushed an update just making the `bare` class do what we need

**Pull request merged**